File: 771-Jewels-and-Stones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
    let nStons = 0;


    for (let stone of stones) {
        if (jewels.indexOf(stone) != -1) nStons++;
    }


    return nStons;
};

File: 771-Jewels-and-Stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        map<char, bool> jewelsMap;
        int count = 0;

        for (char j : jewels) jewelsMap[j] = true;

        for (char s : stones) {
            if (jewelsMap[s]) count++;
        }
        return count;
    }
};